test/unit/gateways/hi_pay_test.rb

```diff
@@ -146,6 +146,19 @@ def test_purchase_with_stored_pm
     end.respond_with(successful_capture_response)
   end
 
+  def test_authorization_string_with_nil_values
+    auth_string_nil_value_first = @gateway.send :authorization_string, [nil, '123456', 'visa']
+    assert_equal '123456|visa', auth_string_nil_value_first
+
+    auth_string_nil_values = @gateway.send :authorization_string, [nil, '123456', nil]
+    assert_equal '123456', auth_string_nil_values
+  end
+
+  def test_authorization_string_with_full_values
+    complete_auth_string = @gateway.send :authorization_string, %w(86786788 123456 visa)
+    assert_equal '86786788|123456|visa', complete_auth_string
+  end
+
   def test_purhcase_with_credit_card; end
```

Comment:
💭 Opinion
I would like to include those cases where you have all of them in nil. What we should expect to get?

Reply:
yes, you are righ, I will add it, thanks

Reply:
done